### PR TITLE
ci: add a go prod stage to deploy tag on gitlab

### PR DESCRIPTION
**Pull request by lionelB**
this PR add a "go prod" stage on travis that trigger a deployement for tagged version

### CODE REVIEW

**Comment by lionelB** [comment]
@douglasduteil can you take a look if everything looks fine ?

**Comment by revolunet** [comment]
I still dont understand why this is not handled directly in gitlab.

we could add a "go prod" stage here which does release+changelog+deploy

**Comment by lionelB** [comment]
@revolunet using travis allow us to bypass account limitation on gitlab. 
About the release + changelog, we could migrate this stage to gitlab.

File: .gitlab-ci/env.sh [hunk 1]
```diff
@@ -2,46 +2,60 @@
...
 export COMMIT_TAG=${COMMIT_TAG:=$CI_COMMIT_TAG}
+export COMMIT=${COMMIT:=$CI_COMMIT_SHA}
+export ENVIRONMENT=${ENVIRONMENT:="dev.factory"};
+export HASH_SIZE=${HASH_SIZE:=7}
 export JOB_ID=${JOB_ID:=$CI_JOB_ID}
-export HASH_SIZE=${HASH_SIZE}
-export ENVIRONMENT=${ENVIRONMENT:="dev"};
-export CLUSTER_NAME="dev.factory";
-BRANCH_NAME_HASHED=$(printf "${BRANCH_NAME}" | sha1sum | cut -c1-${HASH_SIZE} )
+export PROJECT_PATH=${PROJECT_PATH:=$CI_PROJECT_PATH}
```
Comment (lionelB) [inline .gitlab-ci/env.sh:11]: I don't see where it is used

File: .gitlab-ci/env.sh [hunk 2]
```diff
@@ -2,46 +2,60 @@
 
 #
 
-export BRANCH_NAME=${BRANCH_NAME:=$CI_COMMIT_REF_NAME}
-export COMMIT=${COMMIT:=$CI_COMMIT_SHA}
+export BRANCH_NAME=${BRANCH_NAME:=$CI_COMMIT_REF_SLUG}
```
Comment (lionelB) [inline .gitlab-ci/env.sh:5]: what is the difference between `CI_COMMIT_REF_NAME` and `CI_COMMIT_REF_SLUG`

File: .gitlab-ci/env.sh [hunk 3]
```diff
@@ -2,46 +2,60 @@
...
 if [[ -n "${PRODUCTION+x}" ]]; then
-  export ENVIRONMENT="incubateur";
-  export CLUSTER_NAME="incubateur";
-  export BRANCH_HASH_DOT=""
-  export ELASTICSEARCH_HOST=elasticsearch:${ES_PORT};
+  export BRANCH_HASH=prod;
 else
-  export BRANCH_HASH_DOT="${BRANCH_HASH}."
-  export ELASTICSEARCH_HOST=elasticsearch-${BRANCH_HASH}:${ES_PORT};
+  export DOMAIN="${BRANCH_HASH}.${DOMAIN}";
```
Comment (lionelB) [inline .gitlab-ci/env.sh:34]: versions also need a prefix before the domain, no ?
`api.2-3-3.code-du-travail-numerique.incubateur.social.gouv.fr`

File: .gitlab-ci/env.sh (hunk 2 above)
Comment (douglasduteil) [reply .gitlab-ci/env.sh:5]: `CI_COMMIT_REF_NAME` is the raw name 
`CI_COMMIT_REF_SLUG` is the slugify version (with all special char replaced by `-`

Comment (douglasduteil) [reply .gitlab-ci/env.sh:5]: As K8S doesn't like special char I use the slug here

File: .gitlab-ci/env.sh (hunk 1 above)
Comment (douglasduteil) [reply .gitlab-ci/env.sh:11]: Should be use in the standalone version of the github deploy keys scripts

File: .gitlab-ci/env.sh (hunk 3 above)
Comment (douglasduteil) [reply .gitlab-ci/env.sh:34]: yes see below prefixing

**Review by douglasduteil — APPROVED** [review]

**Pull request merged**